azurelinuxagent/common/rdma.py

```diff
@@ -90,3 +96,133 @@ def reboot_system(self):
            the rdma driver cannot be unloaded with rmmod"""
         logger.info('System reboot')
         shellutil.run('shutdown -r now')
+
+dapl_config_paths = [
+    '/etc/dat.conf', '/etc/rdma/dat.conf', '/usr/local/etc/dat.conf']
+
+class RDMADeviceHandler(object):
+
+    """
+    Responsible for writing RDMA IP and MAC address to the /dev/hvnd_rdma
+    interface.
+    """
+
+    rdma_dev = '/dev/hvnd_rdma'
+    device_check_timeout_sec = 120
+    device_check_interval_sec = 1
+
+    ipv4_addr = None
+    mac_adr = None
+
+    def __init__(self, ipv4_addr, mac_addr):
+        self.ipv4_addr = ipv4_addr
+        self.mac_addr = mac_addr
+
+    def start(self):
+        """
+        Start a thread in the background to process the RDMA tasks and returns.
+        """
+        logger.info("RDMA: starting device processing in the background.")
+        threading.Thread(target=self.process).start()
```

Comment:
@rjschwei this is the same behavior as v2.0, if we don't do background, this thread may block up to 2 minutes (so for instance, your PRs need to be executed to set up the RDMA drivers, so the device comes up and I can write the IP to there)... so I think we'll keep it.